tests/integration/test_remoted/test_configuration/test_basic_configuration_connection_invalid_protocol.py

```diff
@@ -0,0 +1,96 @@
+    {'protocol': 'Testing,UDP', 'connection': 'syslog', 'port': '514'},
+    {'protocol': 'TCP,Testing', 'connection': 'syslog', 'port': '514'},
+    {'protocol': 'Testing,Testing', 'connection': 'syslog', 'port': '514'},
+    {'protocol': 'TCP,UDP,Testing', 'connection': 'syslog', 'port': '514'}
+]
+
+configurations = load_wazuh_configurations(configurations_path, "test_basic_configuration_connection",
+                                           params=parameters, metadata=metadata)
+configuration_ids = [f"{x['PROTOCOL']}_{x['CONNECTION']}_{x['PORT']}" for x in parameters]
+
+
+# fixtures
+@pytest.fixture(scope="module", params=configurations, ids=configuration_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+def test_invalid_protocol(get_configuration, configure_environment, restart_remoted):
+    """
+    """
+
+    cfg = get_configuration['metadata']
+    protocol_field = cfg['protocol'].split(',')
+
+    valid_invalid_protocols = remote.get_protocols(protocol_field)
+
+    valid_protocol = valid_invalid_protocols[0]
+    invalid_protocol_list = valid_invalid_protocols[1]
+
+    for invalid_protocol in invalid_protocol_list:
+        log_callback = remote.callback_ignored_invalid_protocol(invalid_protocol)
+        wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                                error_message="The expected error output has not been produced")
+
+    if len(valid_protocol) == 0:
+        log_callback = remote.callback_error_getting_protocol()
+        wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                                error_message="The expected error output has not been produced")
+
+    elif len(valid_protocol) == 1:
+        log_callback = remote.callback_detect_remoted_started(cfg['port'], valid_protocol[0], cfg['connection'])
+        wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                                error_message="The expected error output has not been produced")
+
+    else:
+        used_protocol = 'TCP,UDP'
+        if cfg['connection'] == 'syslog':
+            used_protocol = 'TCP'
+        log_callback = remote.callback_detect_remoted_started(cfg['port'], used_protocol, cfg['connection'])
+        wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                                error_message="The expected error output has not been produced")
+
+    # Check that API query return the selected configuration
+    for field in cfg.keys():
+        api_answer = api.get_manager_configuration(section="remote", field=field)
+        if field == 'protocol':
+            array_protocol = np.array(cfg[field].split(","))
+            assert (array_protocol == api_answer).all(), \
+                "Wazuh API answer different from introduced configuration for protocol value"
+        else:
+            assert cfg[field] == api_answer, "Wazuh API answer different from introduced configuration"
```

Comment:
You don't need to create NumPy arrays to compare them. Maybe using a `lambda` function with map or similar.